server/s3_delete_object_action.cc

```diff
@@ -354,6 +365,14 @@ void S3DeleteObjectAction::add_oids_to_probable_dead_oid_list() {
   next();
 }
 
+void S3DeleteObjectAction::delete_version_metadata() {
+  s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  object_metadata->remove_version_metadata(
```

Comment:
I hope this deletes the metadata from the object version list index table.

Reply:
Yes object_metadata load by parent class, if version id provided then it load from version index else object index